samples/nrf9160/modem_shell/src/location/location_service_lwm2m.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <modem/location.h>
+#include <net/lwm2m_client_utils_location.h>
+
+#include "cloud_lwm2m.h"
+#include "mosh_print.h"
+#include "mosh_defines.h"
+
+/* Location Assistance resource IDs */
+#define GROUND_FIX_SEND_LOCATION_BACK	0
+#define GROUND_FIX_RESULT_CODE		1
+#define GROUND_FIX_LATITUDE		2
+#define GROUND_FIX_LONGITUDE		3
+#define GROUND_FIX_ACCURACY		4
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+void location_service_agps_ext_handle(const struct nrf_modem_gnss_agps_data_frame *agps_req)
+{
+	int err;
+
+	if (!cloud_lwm2m_is_connected()) {
+		mosh_error("LwM2M not connected, can't request A-GPS data");
+		return;
+	}
+
+	location_assistance_agps_set_mask(agps_req);
+
+	while ((err = location_assistance_agps_request_send(cloud_lwm2m_client_ctx_get(), true)) ==
```

Comment:
Is there a change to get stuck here forever if the P-GPS download gets somehow interrupted and never terminated properly?

Reply:
Yes, it could happen if there's a bug in the LwM2M client utils library. I wouldn't be too worried about that in modem_shell, though.

Reply:
@hiltunent Close or comment further.